### PR TITLE
Tests for call semantics

### PR DESCRIPTION
**Pull request by klausler**
Here's some new tests for semantics to exercise the constraints and restrictions that we'll need to check on procedures and their references.  The tests are not enabled.  More to come, too.

### CODE REVIEW

File: test/semantics/call01.f90 [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    res = ''
+  else
+    res = nested()
+  end if
+ contains
+  character(1) function nested
+    !ERROR: assumed-length CHARACTER(*) function can't recurse
+    nested = f14(n-1) ! 15.6.2.1(3)
+  end function nested
+end function
```
Comment (tskeith) [inline test/semantics/call01.f90]: C723 places constraints on a "function name declared with an asterisk type-param-value", not just `CHARACTER(*)` functions. But it seems that other constraints prevent `TYPE(t(*))`, for example. Is that right?

Comment (klausler) [reply test/semantics/call01.f90]: That's exactly right and the Standard should be more specific.  But note that the relevant text in the Standard is in a smaller font: that means that the text pertains only to a deprecated feature.  So there's no point asking for a clarifying edit.

File: test/semantics/call03.f90 [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+    ! ERROR: effective argument associated with INTENT(OUT) dummy must be definable
+    call intentout(3.14159)
+    ! ERROR: effective argument associated with INTENT(OUT) dummy must be definable
+    call intentout(in + 1.)
+    ! ERROR: effective argument associated with INTENT(IN OUT) dummy must be definable
+    call intentinout(in)
+    ! ERROR: effective argument associated with INTENT(IN OUT) dummy must be definable
+    call intentinout(3.14159)
+    ! ERROR: effective argument associated with INTENT(IN OUT) dummy must be definable
+    call intentinout(in + 1.)
```
Comment (tskeith) [inline test/semantics/call03.f90]: Maybe add a test for a parenthesized name:
```
call intentout(x)  ! OK
call intentout((x))  ! ERROR
```

Comment (klausler) [reply test/semantics/call03.f90]: Yes.

File: test/semantics/call07.f90 [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  subroutine test
+    ! ERROR: CONTIGUOUS pointer must be an array
+    real, pointer, contiguous :: a01 ! C830
+    real, pointer :: a02
+    real, target :: a03(10)
+    real :: a04(10) ! not TARGET
+    call s01(a03) ! ok
+    ! ERROR: Effective argument associated with CONTIGUOUS POINTER dummy argument must be simply contiguous
+    call s01(a02)
```
Comment (tskeith) [inline test/semantics/call07.f90]: Wouldn't this just be an error due to rank mismatch?

Comment (klausler) [reply test/semantics/call07.f90]: Right, I omitted the deferred bounds on the declaration.

**Pull request merged**